OpenSim/Simulation/Test/testInitState.cpp

```diff
@@ -146,23 +146,44 @@ void testStates(const string& modelFile)
     ASSERT(max(abs(y1-y2)) > 1e-4);
 }
 
+size_t getMedianOfCurrentRSSOverNSamples(size_t nsamples) {
+    // NOTE: getCurrentRSS() can be unreliable at evaluating the memory usage
+    // of a current process because of caching, shared memory and a litany of 
+    // other reasons including some platform specific details.
+    // Taking an average over nsamples just in case we get a flaky estimate.
+    // A stable estimate of memory usage is necessary so that we do not find
+    // cases where loading model either decreases or has no change in memory
+    // usage, which is impossible.
```

Comment:
Not sure this is the right approach, and I don't see how this change is related to the topic of the PR ("Fix path failure when WrapObject is attached to a PhysicalOffsetFrame"). Concerns:

1. It is still possible for the `size_t mem0 = getMedianOfCurrentRSSOverNSamples(nsamples);` line to be executed immediately before a chunk of memory is freed, in which case the `ASSERT(current_size > mem0, ...);` will be thrown. We have lots of false alarms on CI (e.g., testContactGeometry in Debug mode) and it would be good to avoid creating more of these.

2. Instead of calling `getCurrentRSS()` 10 times in a row, perhaps the entire operation (set `mem0`, load model, compute memory increase) should be repeated.

3. It isn't clear why the median is being computed rather than, say, the maximum (e.g., if memory is freed after the first iteration, `samples[nmedian] < samples[1]` and perhaps `samples[nmedian] < mem0`).

I suggest backing off this commit and investigating in a separate PR.